File: ios/angelia/ReactBridge/AppConfigurationModule.m
//
//  ConfigurationKeys.m
//  angelia
//
//  Created by Mike on 05/12/2017.
//  Copyright © 2017 Facebook. All rights reserved.
//

#import "AppConfigurationModule.h"
#import <React/RCTConvert.h>
#import <React/RCTEventDispatcher.h>
#import <sys/utsname.h>

@implementation AppConfigurationModule

RCT_EXPORT_MODULE();

- (NSDictionary<NSString *, id> *)constantsToExport {
    AppConfigurationModule *keys = [AppConfigurationModule sharedInstance];
    return @{@"apiServer":keys.apiServer,
             @"channel":keys.channelId,
             @"umengAppKey":keys.umengAppKey,
             @"jpushAppKey":keys.jpushAppKey,
             @"codePushKey":keys.codePushKey,
             @"codePushServerURL":keys.codePushServerURL,
             @"socialWechatAppId":keys.socialWechatAppId,
             @"socialWechatAppSecret":keys.socialWechatAppSecret,
             @"socialQQAppId":keys.socialQQAppId,
             @"socialQQAppSecret":keys.socialQQAppSecret,
             @"socialQQAppCallback":keys.socialQQAppCallback,
             @"socialSinaWeiboAppId":keys.socialSinaWeiboAppId,
             @"socialSinaWeiboAppSecret":keys.socialSinaWeiboAppSecret,
             @"socialSinaWeiboAppCallback":keys.socialSinaWeiboAppCallback,
             @"deviceModel":[self deviceModel] // @"iPhone10,3" -> @"iPhone X (CDMA)"  @"iPhone10,6" -> @"iPhone X (GSM)"
             };
}

- (NSString *)deviceModel {
    struct utsname systemInfo;
    uname(&systemInfo);
    NSString *deviceModel = [NSString stringWithCString:systemInfo.machine
                                               encoding:NSUTF8StringEncoding];
    return deviceModel;
}

+ (instancetype)sharedInstance {
    static dispatch_once_t onceToken;
    static AppConfigurationModule *keys;
    dispatch_once(&onceToken, ^{
        keys = [[AppConfigurationModule alloc] init];
    });
    return keys;
}

- (instancetype)init
{
    self = [super init];
    if (self) {
#ifdef DEBUG
        self.apiServer   = @"https://api.wc.kosun.net";
        self.channelId     = @"test";
        self.umengAppKey = @"5a3bbe388f4a9d0a350002be";
        self.jpushAppKey = @"b3f940c3bbc1f7759afc602f";

        self.codePushKey        = @"mVrBpflNHnfN8DVDIsfkPJJWJjhg4ksvOXqog";
        self.codePushServerURL  = @"http://api.codepush.cc/";
        self.codePushAppVersion = @"1.0.0";

        self.socialWechatAppId     = @"wx9ac888ff8ba6a7a2";
        self.socialWechatAppSecret = @"7c56e1a631cd048f002ab4fff7ed109d";

        self.socialQQAppId       = @"";
        self.socialQQAppSecret   = @"";
        self.socialQQAppCallback = @"";

        self.socialSinaWeiboAppId       = @"";
        self.socialSinaWeiboAppSecret   = @"";
        self.socialSinaWeiboAppCallback = @"";
#else
        self.apiServer   = @"todo-apiServer";
        self.channelId     = @"todo-channel";
        self.umengAppKey = @"todo-umengAppKey";
        self.jpushAppKey = @"todo-jpushAppKey";

        self.codePushKey        = @"todo-codePushKey";
        self.codePushServerURL  = @"todo-codePushServerURL";
        self.codePushAppVersion = @"1.0.0";

        self.socialWechatAppId     = @"todo-socialWechatAppId";
        self.socialWechatAppSecret = @"todo-socialWechatAppSecret";

        self.socialQQAppId       = @"todo-socialQQAppId";
        self.socialQQAppSecret   = @"todo-socialQQAppSecret";
        self.socialQQAppCallback = @"todo-socialQQAppCallback";

        self.socialSinaWeiboAppId       = @"todo-socialSinaWeiboAppId";
        self.socialSinaWeiboAppSecret   = @"todo-socialSinaWeiboAppSecret";
        self.socialSinaWeiboAppCallback = @"todo-socialSinaWeiboAppCallback";
#endif
    }
    return self;
}

@end

File: android/app/src/main/java/com/angelia/wancai/module/AppConfigurationModule.java
package com.angelia.wancai.module;

import com.angelia.wancai.MainApplication;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;

import java.util.HashMap;
import java.util.Map;

import javax.annotation.Nullable;

public class AppConfigurationModule extends ReactContextBaseJavaModule {

    private ReactApplicationContext context;

    public AppConfigurationModule(ReactApplicationContext reactContext) {
        super(reactContext);
        context = reactContext;
    }

    @Override
    public String getName() {
        return "AppConfigurationModule";
    }

    @Nullable
    @Override
    public Map<String, Object> getConstants() {
        final Map<String, Object> constants = new HashMap<>();
        constants.put("apiServer", MainApplication.getAppConfiguration().apiServer);
        constants.put("codePushKey", MainApplication.getAppConfiguration().codepushKey);
        constants.put("codePushServerUrl", MainApplication.getAppConfiguration().codepushServerUrl);
        constants.put("channel", MainApplication.getAppConfiguration().channel);
        constants.put("umengAppKey", MainApplication.getAppConfiguration().umengAppKey);
        constants.put("umengAppSecret", MainApplication.getAppConfiguration().umengAppSecret);
        constants.put("jpushAppKey", MainApplication.getAppConfiguration().jpushAppKey);
        // keys
        constants.put("socialWechatAppId", MainApplication.getAppConfiguration().socialWechatAppId);
        constants.put("socialWechatAppSecret", MainApplication.getAppConfiguration().socialWechatAppSecret);
        constants.put("socialQQAppId", MainApplication.getAppConfiguration().socialQQAppId);
        constants.put("socialQQAppSecret", MainApplication.getAppConfiguration().socialQQAppSecret);
        constants.put("socialQQAppCallback", MainApplication.getAppConfiguration().socialQQAppCallback);
        constants.put("socialSinaWeiboAppId", MainApplication.getAppConfiguration().socialSinaWeiboAppId);
        constants.put("socialSinaWeiboAppSecret", MainApplication.getAppConfiguration().socialSinaWeiboAppSecret);
        constants.put("socialSinaWeiboAppCallback", MainApplication.getAppConfiguration().socialSinaWeiboAppCallback);
        return constants;
    }
}
